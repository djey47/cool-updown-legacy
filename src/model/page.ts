export interface MetaOptions {
  refreshSeconds?: number;
}
