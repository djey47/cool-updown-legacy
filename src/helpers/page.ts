import config from 'config';
import { AppConfig, ColorTheme } from '../model/models';
import { THEMES } from '../resources/themes';

export function generatePage(contentsAsHtml: string) {
  return `
  ${generateSharedPageContents()}
  ${contentsAsHtml}
  `;
}

function generateSharedPageContents() {
    const appConfig = config as unknown as AppConfig;
    return `
    ${generateStyles(appConfig.ui?.theme)}
    `;
}

function retrieveThemeRules(theme: ColorTheme) {
  // console.log('page::retrieveThemeRules', { theme });

  if (!theme || !THEMES[theme]) {
    return THEMES.light;
  }

  return THEMES[theme];
}

function generateStyles(theme: ColorTheme) {
  const { fontFamily, backgroundColor, color, linkColor } = retrieveThemeRules(theme);
  const styleRules = `
    body { 
      font-family: ${fontFamily};
      background-color: ${backgroundColor};
      color: ${color};
    }
    body a {
      color: ${linkColor};
    }
  `;
  return `<style type="text/css">${styleRules}</style>`
}
