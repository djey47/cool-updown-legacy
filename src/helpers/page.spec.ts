import { generatePage } from './page';

describe('page helper', () => {
  describe('generatePage function', () => {
    it('should include styles with provided theme', () => {
      // given-when
      const generated = generatePage('<div>contents</div>');

      // then
      expect(generated).toMatchSnapshot();
    });
  });
});
