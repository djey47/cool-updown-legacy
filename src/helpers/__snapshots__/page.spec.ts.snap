// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page helper generatePage function should include styles with provided theme 1`] = `
"
  
    <style type="text/css">
    body { 
      font-family: math;
      background-color: black;
      color: white;
    }
    body a {
      color: gray;
    }
  </style>
    
  <div>contents</div>
  "
`;
